src/command.rs

```diff
@@ -55,25 +57,36 @@ pub fn match_playlist(spotify_client_id: &str, spotify_client_secret: &str, matc
             process::exit(1);
         }
     };
-    let mut match_tracks: Vec<FullTrack> = Vec::new();
-    let mut mismatched_tracks: i32 = 0;
+    let mut matched_tracks: Vec<FullTrack> = Vec::new();
 
     for track in playlist.tracks.values() {
         match spotify_wrapper.exact_track_match(
             &track.name,
             Some(&track.artist),
             track.album.as_ref().map(|x| &**x),
         ) {
-            Ok(spotify_track) => match_tracks.push(spotify_track.clone()),
+            Ok(spotify_track) => matched_tracks.push(spotify_track.clone()),
             Err(e) => match e.kind {
-                ErrorKind::MultipleResults(_) | ErrorKind::NoResults => mismatched_tracks += 1,
+                ErrorKind::MultipleResults(_) | ErrorKind::NoResults => (),
```

Comment:
I'll have to understand more of the code but it seems like you're just squashing the error here?

Reply:
Good point. I intentionally was returning an error because I wanted to handle it. Given that I am no longer doing anything with these errors, it would be better to return an option from the method than an error.